File: routes/admin/goodsdel.js

var res = require('./lib/resp.js');
var goods = require("./models/goods.js");
var dbUtils = require('./models/dbUtils.js');



if (!session.get("isAdmin")) {
  return res.redirect('/routes/admin/login.js');
}

var oid = request.formValue('oid');
goods.delByOid(oid);
res.redirect('./goodslist.js');

File: routes/admin/goodsdel.go
package admin

import (
	"github.com/zengming00/go-testShop/framework"
)

func GoodsDel(ctx *framework.HandlerContext) {
	var isAdmin = false
	if v, ok := ctx.GetSessionVal("isAdmin"); ok {
		isAdmin = v.(bool)
	}
	if !isAdmin {
		ctx.Redirect("/routes/admin/login.go")
	}

	var oid = ctx.R.FormValue("oid")
	ctx.Goods.DelByOid(oid)
	ctx.Redirect("./goodslist.go")
}
